app/assets/javascripts/components/client/EditClient.jsx

```diff
@@ -2,20 +2,78 @@
 class EditClient extends React.Component {
 
   constructor(props) {
-    super();
-    this.state = {
-      users: [],
+        super();
+        this._update = this._update.bind(this);
+        this._handleSubmit = this._handleSubmit.bind(this);
+        this.state = {
+            client: {},
+            showForm: true,
+            message: '',
+            error: ''
+        };
     }
-  }
 
   componentDidMount() {
+    Requester.get('/api/clients/' + this.props.client.id).then((data) => {
+      this.setState({client: data});
+    });
+  }
 
-    Requester.get('/api/users/').then((data) => {
-      this.setState({users: data});
+  _update = (name, val) => {
+    const client = this.state.client;
+    client[name] = val;
+    this.setState({
+      client: client
     });
   }
 
+  _handleChange = (e) => {
+    const client = this.state.client;
+    client[$(e.target).attr("name")] = $(e.target).val();
+  }
+
+  _safeTrim = (val) => {
+      if (val != undefined) {
+          return val.trim()
+      }
+      return val
+  }
+
+  _handleSubmit = (e) => {
+      e.preventDefault();
+      const payload = {
+          client: this.state.client
+      };
+
+      Requester.update('/api/clients/' + this.state.client.id, payload).then((data) => {
+          this.setState({
+              "message": data.message,
+              "error": data.error,
+              "showForm": false
+          });
+      });
+
+      window.location = '/clients/' + this.state.client.id + '/profile';
```

Comment:
:eyes

Reply:
👀

Reply:
lol, should i change this?? haha. is there a better way

Reply:
just giving @amyshan a hard time dw